### DISCUSSION TITLE
How do I install the PL2303 driver to connect the X-Dream Bike to the X-Dream software?

### DISCUSSION THREAD

**Discussion by drmason789** [xdream-biking/trixter-xdream-bike#6] [Q&A] [answered]
The X-Dream bike is connected to the computer by means of a USB cable, but it communicates with the computer using a serial to USB converter, a “Prolific(TM) PL-2303” to be more precise.

This requires the installation of a driver.

**Comment by drmason789** [comment, marked as answer]
In the Trixter\Live folder there should be an executable called:

	PL-2303 Driver Installer.exe

![image](https://user-images.githubusercontent.com/29954900/143244602-830764d4-9554-4396-ad72-022a1f9c5104.png)

Double click file to run it and install the driver.
Once it’s done, navigate to the Device Manager:
![image](https://user-images.githubusercontent.com/29954900/143244699-a29610fb-0291-4031-b119-96ba41a65eb9.png)

Under Ports (COM and LPT) there should now be “Prolific USB-to-Serial Comm Port (COM-x)” where x is a number. This number needs to be 2 (unless a different port is configured in the Trixter application, in whichever one of the main*.txt files it uses for configuration), but it probably isn’t.
![image](https://user-images.githubusercontent.com/29954900/143244821-9c3fb86d-e8ca-41f6-87c7-be43e39d3204.png)
If it does not say COM2, right click the device and select Properties.

![image](https://user-images.githubusercontent.com/29954900/143244857-a409d339-024b-4c3c-aa48-a352f4c1e747.png)

Then configure the port as follows:

![image](https://user-images.githubusercontent.com/29954900/143244886-b376cca0-cefa-4aaa-959f-feaca4dec7be.png)

1.	Select the Port Settings tab
2.	Click the Advanced button
3.	Set the COM port number to COM2
4.	Click OK
5.	Set the Bits Per Second to 115200 (the software that uses the port should override this at runtime).
6.	Switch to the Power Management tab:
![image](https://user-images.githubusercontent.com/29954900/143244937-b0dd8b52-6408-4418-a13d-2550453b0d49.png)
7.	Disable “Allow the computer to turn off this device to save power”.
8.	Click OK on the properties dialog.

It may warn you that another application has already opened the port or some other error. Ignore this. Do check that other COM ports in under “Ports (COM & LPT) are not configured to COM2 though – change them to something else or disable them if they are.
Restart the machine and return to the device manager.

A problem you might encounter here is an update of the driver, either automatically by Windows, a person doing it manually, or any driver update utility. This will likely make it stop working. It can look like this:

![image](https://user-images.githubusercontent.com/29954900/143245027-d892bb7c-1d33-47dc-aeca-6b72de800baa.png)

If you see this, refer to discussion #8 .

**Reply by dermyhughes** [reply]
Just want to thank you for this. I just swapped out a dead CMOS battery in the original Dell PC and lost all connection to the bike. I was out of ideas when I found this repo and your comment about it needing to be on COM2.
